taking-turns.js

```diff
@@ -0,0 +1,241 @@
+          lastHighLow = "lower";
+        }
+      }
+      cheatDetector(guess, lastGuess, allGuesses);
+    }
+  }
+  totalComputerTries += tries;
+  totalComputerWins++;
+  nextPlayer = "user";
+  nextGame();
+}
+
+//The computer picks a number and the user guesses
+async function userGuesses() {
+  let userGuesses = [];
+  let tries = 1;
+  console.log("\nTell me the range you want and I'll pick a number.\n");
+  
+  //Get the min value as an int
+  console.log("Please enter the lower limit: ");
+  minNumber = await getNumber();
+  
+  //Get the max value as an int
+  console.log("Please enter the upper limit: ");
+  maxNumber = await getNumber();
+  
+  let secretNumber = randomInteger(minNumber, maxNumber);
+  
+  console.log("Okay I picked a number. Good luck guessing!");
+  //Uncomment to see what the computer is picking for it's number
+  //console.log("My secret number is: " + secretNumber); 
+
+  //Start guessing
+  let guess = await ask("What's your guess? ");
+  
+  while (true) {
+    userGuesses.push(guess);
+    if (guess < secretNumber) {
+      console.log("Nope. My number is higher.");
+    } else if (guess > secretNumber) {
+      console.log("Nope. My number is lower.");
+    } else if (isNaN(guess)) {
+      console.log("That wasn't a number! But I'm counting it as a guess.");
+    } else {
+      console.log("You got it!\nYour number of guesses: " + tries);
+      console.log("Your guesses were: " + userGuesses);
+      break;
+    }
+    tries++;
+    guess = await ask("What's your guess? ");
+  }
+  totalUserTries += tries;
+  totalUserWins++;
+  nextPlayer = "computer";
+  nextGame();
+}
+
+//This function asks the user if they want to keep playing and, if so, checks who guesses next
+async function nextGame() {
+  console.log("\nDo you want to play again? (Y/N): ")
+  playAgain = await getYesNo();
+  if (playAgain==="y") {
+    nextPlayer === "user" ? userGuesses() : computerGuesses();
+  } else {
+    //assumes no
+    endGame();
+  }
+}
+//Text and data for the end of the game
+function endGame() {
+  let averageComputerGuesses = totalComputerTries / totalComputerWins;
+  let averageUserGuesses = totalUserTries / totalUserWins;
+
+  console.log(`\nNumber of games that I guessed correctly: ${totalComputerWins}`);
+  console.log(`My average number of guesses: ${averageComputerGuesses}`);
+
+  console.log(`\nNumber of games that you guessed correctly: ${totalUserWins}`);
+  console.log(`Your average number of guesses: ${averageUserGuesses || 0}`);
+
+  if (averageComputerGuesses > averageUserGuesses) {
+    console.log("\nYou were better at guessing! Great job!");
+  } else if (averageComputerGuesses < averageUserGuesses) {
+    console.log("\nI was better at guessing this time. Better luck next time.");
+  } else {
+    console.log("\nWe were evenly matched! Well played!");
+  }
+  console.log("\nIt was nice playing with you. Goodbye.");
+  process.exit();
+}
+
```

Comment:
Consider keeping all the helper functions (endGame, nextGame, getNumber, getYesNo, getHighlow, randomInteger, and cheatDetector) in one location to improve readability